source/nodejs/adaptivecards-designer/src/designer-peers.ts

```diff
@@ -2424,16 +2544,24 @@ export class ImagePeer extends TypedCardElementPeer<Adaptive.Image> {
 export class MediaPeer extends TypedCardElementPeer<Adaptive.Media> {
     static readonly altTextProperty = new StringPropertyEditor(Adaptive.Versions.v1_1, "altText", "Alternate text", true);
     static readonly posterUrlProperty = new StringPropertyEditor(Adaptive.Versions.v1_1, "posterUrl", "Poster URL", true);
-    static readonly sourcesProperty = new NameValuePairPropertyEditor(
+    static readonly sourcesProperty = new InnerStructPropertyEditor(
         Adaptive.Versions.v1_1,
         "sources",
-        "url",
-        "mimeType",
-        (name: string, value: string) => { return new Adaptive.MediaSource(name, value); },
-        "URL",
-        "MIME type",
+        ["url", "mimeType"],
+        (innerProperties: string[]) => { return new Adaptive.MediaSource(innerProperties[0], innerProperties[1]); },
+        ["URL", "MIME type"],
+        2,
```

Comment:
I think this can be simplified - check out this (non-specific) [snippet I was toying with](https://www.typescriptlang.org/play?#code/C4TwDgpgBAsiAKBDAlgJygXigbwFBQKgDtEBbCALigGdhVkiBzAGn0IDdEAbAVwgH4qtek1wBfXLlCRYIAEKJGmHGwIBtEuSF0GSgLpU4SNOMlcIwKADMA9jaiH5i5XkJQARD1Rd3VV28JNSg8AVQAlABl3VSgxVjd3UmRyABVwCF8VAMCyYPcYAEkYAFEoNMhotwkxAG5JWxs1T293PQA6Tl5oLHcAC2BgMGoKAHoR6htyYF7ddzrcAHkAIwArCABjYDaAawgQagAKBoBKNttUYsR13oPdkEwAPih1myIJ8zauG0YDgAMAEmwdzEVEBDTUd3aQTEUAOnT4oOw4MhHW4fCg-H4HiIPC4PjEx1+x2OdSAA)

Reply:
That's a lot better, implemented! Thank you